### PR TITLE
Introduce new structure for running TCK

### PR DESCRIPTION
**Pull request by Mats-SX**
This PR does several things:
- Adds ANTLR dependency, grammar, and generated parser, for parsing the expected results from the TCK's `.feature` files
- Implements a parsing layer that converts expected results into Neo4j objects
- Adds support for the new framework in the Cucumber implementation (`GlueSteps`)
- Adds a new feature `Literals` that uses the new framework

### CODE REVIEW

File: community/cypher/compatibility-suite/pom.xml [hunk 1]
```diff
@@ -73,6 +73,12 @@
 
     <!-- shared versions are defined in the parent pom -->
 
+    <dependency>
+      <groupId>org.antlr</groupId>
+      <artifactId>antlr4</artifactId>
+      <version>4.5.1</version>
+    </dependency>
```
Comment (davidegrohmann) [inline community/cypher/compatibility-suite/pom.xml]: maybe test only?

**Comment by boggle** [comment]
This is really nice

File: community/cypher/compatibility-suite/src/main/java/cypher/feature/parser/CypherValuesCreator.java [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.Stack;
+
+import org.neo4j.graphdb.Label;
+import org.neo4j.graphdb.RelationshipType;
+
+class CypherValuesCreator extends FeatureResultsBaseListener
+{
+    private Stack<Object> workload;
```
Comment (pontusmelke) [inline community/cypher/compatibility-suite/src/main/java/cypher/feature/parser/CypherValuesCreator.java]: Stacks are sort of deprecated right? Use `Deque` instead.

**Comment by Mats-SX** [comment]
@boggle Melded license headers into relevant commits.

**Pull request merged**